packages/x-bundle/src/executors/database.ts

```diff
@@ -149,29 +151,47 @@ export function CheckDocumentExists<T>(
 /**
  * Inserts the document in the required collection.
  * @param collectionClass
+ * @param options contains the following arguments :
  * @param field The field from GRAPHQL arguments
  * @param extend Possibly extend the document, such as adding additional fields like a userId or whatever you wish.
+ * @param deepSync To deepSync all document if true, or deepsync on focusing certain fields as string[] or string
  * @returns
  */
 export function ToDocumentInsert<T>(
   collectionClass: Constructor<Collection<T>>,
-  field = "document",
-  extend?: (document: any, ctx: IGraphQLContext) => void | Promise<void>
+  options?: InsertExecutorOptions
 ) {
   return async function (_, args, ctx, ast) {
     const collection: Collection = ctx.container.get(collectionClass);
-    const document = args[field];
-    if (extend) {
-      await extend(document, ctx);
+    let document = args[options.field || "document"];
+    if (options.extend) {
+      await options.extend(document, ctx);
     }
+    if (options?.deepSync) {
+      if (typeof options.deepSync !== "boolean") {
+        const deepSyncFields = {};
+        (Array.isArray(options.deepSync)
+          ? options.deepSync
```

Comment:
So we are specifying which fields we're going to deep sync?  How does this help us, since document[key] already exists and then you merge it below, won't it be like that everytime?